File: JobQuestFrontend/src/app/models/job-detail-info-model.ts
import { Company } from "./companyModel/company";

export class JobDetailInfoModel {
    public jobId: number;
    public jobName: string;
    public expReq: string;
    public location: string;
    public postedAgo: string;
    public noOfApplicant: number;
    public jobDesc: string;
    public role: string;
    public industryType: string;
    public department: string;
    public empType: string;
    public roleCategory: string;
    public education: string;
    public keySkills: string;
    public salRangeMin: number;
    public salRangeMax: number;
    public workMode: string;
    public compType: string;
    public compId: Company;
    public jobActive: boolean;

    constructor(
        jobId?: number,
        jobName?: string ,
        expReq?: string ,
        location?: string ,
        postedAgo?: string ,
        noOfApplicant?: number,
        jobDesc?: string ,
        role?: string ,
        industryType?: string ,
        department?: string ,
        empType?: string ,
        roleCategory?: string ,
        education?: string ,
        keySkills?: string ,
        salRangeMin?: number,
        salRangeMax?: number,
        workMode?: string ,
        compType?: string ,
        compId?: Company, // Assuming default constructor is available
        isJobActive?: boolean
    ) {
        this.jobId = jobId ?? 0;
        this.jobName = jobName ?? '';
        this.expReq = expReq ?? '';
        this.location = location ?? '';
        this.postedAgo = postedAgo ?? '';
        this.noOfApplicant = noOfApplicant ?? 0;
        this.jobDesc = jobDesc ?? '';
        this.role = role ?? '';
        this.industryType = industryType ?? '';
        this.department = department ?? '';
        this.empType = empType ?? '';
        this.roleCategory = roleCategory ?? '';
        this.education = education ?? '';
        this.keySkills = keySkills ?? '';
        this.salRangeMin = salRangeMin ?? 0;
        this.salRangeMax = salRangeMax ?? 0;
        this.workMode = workMode ?? '';
        this.compType = compType ?? '';
        this.compId = compId ?? new Company();
        this.jobActive = isJobActive ?? true;
    }
}

File: JobQuestRestBackend/src/main/java/com/cybage/boot/models/JobDetailInfoModel.java
package com.cybage.boot.models;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Table(name = "jobDetails")
public class JobDetailInfoModel {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int jobId;
	private String jobName;
	private String expReq;
	private String location;
	private String postedAgo;
	private int noOfApplicant;
	private String jobDesc;
	private String role;
	private String industryType;
	private String department;
	private String empType;
	private String roleCategory;
	private String education;
	private String keySkills;
	private double salRangeMin;
	private double salRangeMax;
	private String workMode;
	private String compType;
	private boolean isJobActive;

	@ManyToOne
	@JoinColumn(name = "compId")
	@JsonIgnore
	private CompanyInfoModel compId;
}